sotodlib/mapmaking/utilities.py

```diff
@@ -533,3 +535,39 @@ def import_optional(module_name):
         return module
     except:
         return None
+
+Base = declarative_base()
+
+class AtomicInfo(Base):
+    __tablename__ = "atomic"
+
+    obs_id: Mapped[str] = mapped_column(primary_key=True)
+    telescope: Mapped[str] = mapped_column(primary_key=True)
+    freq_channel: Mapped[str] = mapped_column(primary_key=True)
+    wafer: Mapped[str] = mapped_column(primary_key=True)
+    ctime: Mapped[int] = mapped_column(primary_key=True)
+    split_label: Mapped[str] = mapped_column(primary_key=True)
```

Comment:
Is your aim here to have these be a composite primary key in the database?

Reply:
Also why use ctime instead of a datetime object?

Reply:
yes, these are the things that combined define a unique atomic map. I thought a primary key was something that uniquely defines a row. Therefore, I made all these primary keys. ctime is just our tradition, we define observations by their ctime.

Reply:
Yes, that's correct. You can certainly do this. One other option is to have a unique ID as the primary key, but if you are certain that _there will never be rows in the database with a duplicate value for these combined items_ then it is fine.

Reply:
Ok. If I try to add another row to the database with the same set of these primary keys, it will error, right? That should never happen because these same keys are the ones used to check for existing maps at the beginning of the mapmaker. That way you don't re-map and existing map.